parse_test.go

```diff
@@ -0,0 +1,192 @@
+package github
+
+import (
+	"flag"
+	"os"
+	"strconv"
+	"testing"
+)
+
+var (
+	tmpKeyFile string = "/tmp/key.pem"
+)
+
+func prepare() {
+	instId = 12345
+	appId = 9876543
+	os.WriteFile(tmpKeyFile, testPrivateKey, 0644)
+}
+
+func cleanup() {
+	os.Remove(tmpKeyFile)
+	os.Args = []string{}
+	os.Setenv("GITHUB_APP_ID", "")
+	os.Setenv("GITHUB_INST_ID", "")
+	os.Setenv("GITHUB_KEY_PATH", "")
+	os.Setenv("GITHUB_KEY_VALUE", "")
+}
+
+func setArgs(args []string) {
+	os.Args = args
+	flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+}
+
+func TestPassedKeyPath(t *testing.T) {
+	prepare()
+	defer cleanup()
+
+	setArgs([]string{"binary", "-i", strconv.FormatInt(instId, 10), "-a", strconv.FormatInt(appId, 10), "-k", tmpKeyFile})
+	cfg, err := ParseParameters()
+	if err != nil {
+		t.Fatalf("Error validating parameters")
+	}
+	if cfg.ApplicationID != appId {
+		t.Errorf("App ID: expect: %v, got: %v", appId, cfg.ApplicationID)
+	}
+	if cfg.InstallationID != instId {
+		t.Errorf("Installation ID: expect: %v, got: %v", instId, cfg.InstallationID)
+	}
+	// Read the content of the key file and compare with the parsed private key
+	keyContent, err := os.ReadFile(tmpKeyFile)
+	if err != nil {
+		t.Fatalf("Failed to read key file: %v", err)
+	}
+	if string(cfg.PrivateKey) != string(keyContent) {
+		t.Errorf("Private Key: expected %s, got %s", string(keyContent), string(cfg.PrivateKey))
+	}
+}
+
+func TestNoAppIDParameter(t *testing.T) {
+	prepare()
+	defer cleanup()
+
+	setArgs([]string{"binary", "-i", strconv.FormatInt(instId, 10), "-k", tmpKeyFile})
+	_, err := ParseParameters()
+	if err != nil && err.Error() != noAppIdError {
+		t.Errorf("expect: %q, got: %q", noAppIdError, err.Error())
+	}
+}
+
+func TestNoInstIDParameter(t *testing.T) {
+	prepare()
+	defer cleanup()
+
+	setArgs([]string{"binary", "-a", strconv.FormatInt(appId, 10), "-k", tmpKeyFile})
+	_, err := ParseParameters()
+	if err != nil && err.Error() != noInstIdError {
+		t.Errorf("expect: %q, got: %q", noInstIdError, err.Error())
+	}
+}
+
+func TestNoKeyPathParameter(t *testing.T) {
+	prepare()
+	defer cleanup()
+
+	setArgs([]string{"binary", "-a", strconv.FormatInt(appId, 10), "-i", strconv.FormatInt(instId, 10)})
+	_, err := ParseParameters()
+	if err != nil && err.Error() != noKeyError {
+		t.Errorf("expect: %q, got: %q", noKeyError, err.Error())
+	}
+}
+
+func TestNoAppIDEnvVar(t *testing.T) {
+	prepare()
+	defer cleanup()
+
+	setArgs([]string{"binary"})
+	os.Setenv("GITHUB_INST_ID", strconv.FormatInt(instId, 10))
+	os.Setenv("GITHUB_KEY_PATH", string(tmpKeyFile))
+	os.Setenv("GITHUB_KEY_VALUE", string(testPrivateKey))
+	_, err := ParseParameters()
+	if err != nil && err.Error() != noAppIdError {
+		t.Errorf("expect: %q, got: %q", noAppIdError, err.Error())
+	}
+}
+
+func TestNoInstIDEnvVar(t *testing.T) {
```

Comment:
This test and `TestNoAppIDEnvVar()` are messing with me.

They both have an _identical_ setup, yet their asserts are different (confused).  Could you combine them an add two asserts?

Another question is the use of _both_ `GITHUB_KEY_PATH` _and_ `GITHUB_KEY_VALUE`.  Shouldn't these parameters be mutually exclusive?  I think it would be a better reflection of reality if _either_ of these parameters were set... then create a test to explore the behavior if *both* are set, but it shouldn't be the "norm".